api/handlers.go

```diff
@@ -816,21 +817,26 @@ func SearchHandler(c *gin.Context) {
 
 	timeoutForHighlight := viper.GetDuration("elasticsearch.timeout-for-highlight")
 
-	res, err := se.DoSearch(
-		context.TODO(),
-		query,
-		sortByVal,
-		from,
-		size,
-		preference,
-		checkTypo,
-		true,
-		true,
-		true,
-		timeoutForHighlight,
-	)
+  var res *search.QueryResult
+  if res, err = searchCache.Get(query, sortByVal, from, size); err == nil && res == nil {
```

Comment:
If we get an error from the cache, which is not `ErrNotFound` I think we should try elastic nonetheless. The cache is not critical.